tools/json2dbc.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python3
+
+import json
+
+import argparse
+import dataclasses
+import textwrap
+import typing
+
+
+@dataclasses.dataclass
+class SignalValue:
+    name: str
+    value: int
+
+    @staticmethod
+    def from_json(key, value):
+        return SignalValue(value, int(key))
+
+    def to_dbc(self):
+        return f'{self.value} "{self.name}"'
+
+
+@dataclasses.dataclass
+class SignalValueDefinition:
+    can_id: int
+    signal_name: str
+    signal_values: typing.List[SignalValue]
+
+    def to_dbc(self):
+        value_table_definition = ' '.join(signal_value.to_dbc() for signal_value in self.signal_values)
+        return f'VAL_ {self.can_id} {self.signal_name} {value_table_definition};'
+
+
+@dataclasses.dataclass
+class Signal:
+    name: str
+    bit_start: int
+    length: int
+    little_endian: bool
+    signed: bool
+    value_type: str
+    factor: float = 1.0
+    offset: float = 0.0
+    min: float = 0.0
+    max: float = 0.0
+    unit: str = ""
+    values: typing.List[SignalValue] = dataclasses.field(default_factory=list)
+
+    @staticmethod
+    def from_json(key, value):
+        values = [SignalValue.from_json(key, value) for key, value in value.get('enums', {}).items()]
+        return Signal(
+            key,
+            value['bit_start'],
+            value['length'],
+            value['little_endian'],
+            value['signed'],
+            value['value_type'],
+            value['factor'],
+            value['offset'],
+            value['min'],
+            value['max'],
+            value.get('unit', ""),
+            values,
+        )
+
+    def to_dbc(self):
+        little_endian = 1 if self.little_endian else 0
+        signed = '-' if self.signed else '+'
+        return f'   SG_ {self.name} : {self.bit_start}|{self.length}@{little_endian}{signed} '\
+            f'({self.factor:g},{self.offset:g}) [{self.min:g}|{self.max:g}] "{self.unit}" Vector__XXX'
+
+
+@dataclasses.dataclass
+class Message:
+    can_id: int
+    name: str
+    sender: str
+    length: int  # in bytes
+    signals: typing.List[Signal]
+
+    @staticmethod
+    def from_json(key, value):
+        signals = []
+        for _key, _value in value.get('signals', {}).items():
```

Comment:
The `_` prefix in Python is used for unused variables and not to avoid name collision ;).